src/sdl2_compat.h

```diff
@@ -194,7 +339,7 @@ typedef enum
   SDL2_SYSWM_RISCOS
 } SDL2_SYSWM_TYPE;
 
-struct SDL2_SysWMinfo
+struct SDL_SysWMinfo
```

Comment:
Can we avoid the typedefs and includes by making `SDL_SysWMinfo` use void pointers instead of typedefs? Or are these important?
It looks like `SDL_GetWindowWMInfo` is simply forwarding pointers and ints.

Reply:
I think we can. It will be a bit ugly, but the SDL2 syswm struct should be forever frozen in time and hopefully this code will never have to be touched again.

Reply:
And you already did the hard work of adding the types. So it would be a waste of reverting this.

Reply:
>  It will be a bit ugly,

It did get ugly:
```diff
+      wminfo->info.x11.window = (unsigned long)(uintptr_t)SDL3_GetProperty(props, "SDL.window.x11.window");
```

Reply:
Losing 30 seconds of copy-paste from the original headers isn't a big deal :)

I like this much better without all of the `#ifdef` mess anyways.  We just need to keep the bit for the Cocoa and UIKit auto refcounting.